packages/task/src/browser/task-service.ts

```diff
@@ -1059,11 +1067,7 @@ export class TaskService implements TaskConfigurationClient {
         terminal.sendText(selectedText);
     }
 
-    async attach(terminalId: number, taskId: number): Promise<void> {
-        // Get the list of all available running tasks.
-        const runningTasks: TaskInfo[] = await this.getRunningTasks();
-        // Get the corresponding task information based on task id if available.
-        const taskInfo: TaskInfo | undefined = runningTasks.find((t: TaskInfo) => t.taskId === taskId);
+    async attach(terminalId: number, taskInfo: TaskInfo): Promise<number | void> {
```

Comment:
Why change the signature if all you ever use on the `taskInfo` is its `taskId` field, which is what was there before?

It is also preferable to not break public API if we can avoid it.

Reply:
`if (TaskOutputPresentation.shouldAlwaysRevealTerminal(taskInfo.config)) {
                if (TaskOutputPresentation.shouldSetFocusToTerminal(taskInfo.config)) { /`
`taskInfo` is used [here](https://github.com/EstherPerelman/theia/blob/19239455b0a042867782b85ee2c6f09cd3e0257a/packages/task/src/browser/task-service.ts#L1072-L1079)...

Reply:
You're right, I was focusing on added code only.